frappe/public/js/frappe/ui/messages.js

```diff
@@ -167,6 +167,10 @@ frappe.msgprint = function(msg, title, is_minimizable) {
 					method: data.primary_action.server_action,
 					args: {
 						args: data.primary_action.args
+					},
+					callback: (r) => {
+						// close msgprint dialog box on primary action
+						frappe.hide_msgprint();
```

Comment:
This should be handled by the developer. Hiding the dialog after primary action is not always expected.